endpoint.cpp

```diff
@@ -95,7 +95,7 @@ Endpoint::~Endpoint()
     free(tx_buf.data);
 }
 
-int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid, int *target_compid)
+int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid)
```

Comment:
> One system id can have several components ids.

In reality we may have multiple systemdid with multiple compid on the same Endpoint.  When doing the route we decide we would assume only one sysid + compid per endpoint.

Have you seem multiple compids in a single endpoint and this is hurting what you are doing here?  We may need to go the other direction and add support for more sysid/compid on a single endpoint rather than removing compid.